lib/Dialect/Arith/Conversions/ArithToCGGI/ArithToCGGI.cpp

```diff
@@ -122,17 +127,49 @@ static Value materializeTarget(OpBuilder& builder, Type type, ValueRange inputs,
                                Location loc) {
   assert(inputs.size() == 1);
   auto inputType = inputs[0].getType();
-  if (!isa<IntegerType>(inputType))
+
+  bool typeCheck = llvm::TypeSwitch<Type, bool>(inputType)
+                       .Case<IntegerType>([](auto type) { return true; })
+                       .Case<ShapedType>([](auto type) {
+                         return isa<IntegerType>(type.getElementType());
+                       })
+                       .Default([](auto type) { return false; });
```

Comment:
You can use getElementOrSelf and then check if that's an integer type

(!isa<IntegerType>(getElementTypeOrSelf(inputType))